packages/loader/container-loader/src/containerStorageAdapter.ts

```diff
@@ -109,15 +109,20 @@ export class ContainerStorageAdapter
 			this.logger,
 		));
 
-		this._summarizeProtocolTree =
-			this._summarizeProtocolTree ?? service.policies?.summarizeProtocolTree;
-		if (this.summarizeProtocolTree) {
-			this.logger.sendTelemetryEvent({ eventName: "summarizeProtocolTreeEnabled" });
-			this._storageService = new ProtocolTreeStorageService(
-				retriableStorage,
-				this.addProtocolSummaryIfMissing,
-			);
-		}
+		// A storage service wrapper which intercept calls to uploadSummaryWithContext and ensure they include
+		// the protocol summary, provided single-commit summary is enabled.
+		this._storageService = new ProtocolTreeStorageService(
```

Comment:
Why are we inverting the logic here? I think we should only wrap inside the ProtocolTreeStorageService if we want to summarize protocol tree and not check inside the ProtocolTreeStorageService if it is enabled or not.

Reply:
This new change is implemented so that we query for the service policies when needed. This is will give drivers enough time to establish socket connection, so that the connection's supportedFeatures can be used if needed. Just like in this PR.